protocol/communities/manager.go

```diff
@@ -1425,8 +1426,9 @@ func (m *Manager) HandleCommunityEventsMessage(signer *ecdsa.PublicKey, message
 	}
 
 	return &CommunityResponse{
-		Community: changes.Community,
-		Changes:   changes,
+		Community:      changes.Community,
+		Changes:        changes,
+		RequestsToJoin: additionalCommunityResponse.RequestsToJoin,
```

Comment:
Yeah I thought of that too.

I didn't find a function to merge the community responses, but yeah it would be better.

I could also just return `[]*RequestToJoin` instead of `*CommunityResponse` like you said, but I didn't know if it might change in the future. `handleAdditionalAdminChanges` right now only handles request to join, but maybe later it would handle some more? I guess in that case, we can just add a new return type.

Reply:
Right, I believe I'll need to propagate also handled permissions and `CommunityResponse` would be a good fit to store that info. Having a Merge method for `CommunityResponse`  would simplify. Here is an example of Merge for `MessengerResponse`: https://github.com/status-im/status-go/blob/3a41a2550aa28fce8c53746f9bf9168edf13beba/protocol/messenger_response.go#L294

PS. I'll implement it in the scope of my PR.